test/powershell/Host/TabCompletion/TabCompletion.Tests.ps1

```diff
@@ -935,6 +935,110 @@ ConstructorTestClass(int i, bool b)
         $res.CompletionMatches[0].CompletionText | Should -BeExactly '$TestVar1'
     }
 
+    It 'Should include help message in parameter tool tip by direct match' {
+        Function Test-Function {
+            param (
+                [Parameter(HelpMessage = 'Some help message')]
+                $Foo
+            )
+        }
+
+        $expected = '[Object] Foo - Some help message'
+        $Script = 'Test-Function -Foo'
+        $res = (TabExpansion2 -inputScript $Script).CompletionMatches
+        $res.Count | Should -Be 1
+        $res.CompletionText | Should -BeExactly '-Foo'
+        $res.ToolTip | Should -BeExactly $expected
+    }
+
+    It 'Should include help message in parameter tool tip by multiple match' {
+        Function Test-Function {
+            param (
+                [Parameter(HelpMessage = 'Some help message')]
+                $Foo,
+
+                $Bar
+            )
+        }
+
+        $expected = '[Object] Foo - Some help message'
+        $Script = 'Test-Function -'
+        $res = (TabExpansion2 -inputScript $Script).CompletionMatches | Where-Object CompletionText -eq '-Foo'
+        $res.Count | Should -Be 1
+        $res.CompletionText | Should -BeExactly '-Foo'
+        $res.ToolTip | Should -BeExactly $expected
+    }
+
+    It 'Should retrieve help message from dynamic parameter' {
+        Function Test-Function {
+            [CmdletBinding()]
+            param ()
+            dynamicparam {
+                $attr = [System.Management.Automation.ParameterAttribute]@{
+                    HelpMessage = "Howdy partner"
+                }
+                $attrCollection = [System.Collections.ObjectModel.Collection[System.Attribute]]::new()
+                $attrCollection.Add($attr)
+
+                $dynParam = [System.Management.Automation.RuntimeDefinedParameter]::new('Foo', [string], $attrCollection)
+
+                $paramDictionary = [System.Management.Automation.RuntimeDefinedParameterDictionary]::new()
+                $paramDictionary.Add('Foo', $dynParam)
+                $paramDictionary
+            }
+
+            end {}
+        }
+
+        $expected = '[string] Foo - Howdy partner'
+        $Script = 'Test-Function -'
+        $res = (TabExpansion2 -inputScript $Script).CompletionMatches | Where-Object CompletionText -eq '-Foo'
+        $res.Count | Should -Be 1
+        $res.CompletionText | Should -BeExactly '-Foo'
+        $res.ToolTip | Should -BeExactly $expected
+    }
+
+    It 'Should have type and name for parameter without help message' {
+        Function Test-Function {
+            param (
+                [Parameter()]
+                $WithAttribute,
+
+                $WithoutAttribute
```

Comment:
A bit confuse - look like an attribute in script below:
```suggestion
                [Parameter()]
                $WithAttributeParam,

                $WithoutAttributeParam
```

Reply:
I'll probably update to `$WithParamAttribute` and `$WithoutParamAttribute`

Reply:
Done with the changes mentioned above.